scripts/install

```diff
@@ -79,40 +76,34 @@ ynh_script_progression --message="Configuring PHP-FPM..." --weight=2
 
 # Create a dedicated PHP-FPM config
 ynh_add_fpm_config
-phpversion=$(ynh_app_setting_get --app=$app --key=phpversion)
 
 #=================================================
-# GENERIC FINALIZATION
-#=================================================
-# SECURE FILES AND DIRECTORIES
+# NGINX CONFIGURATION
 #=================================================
+ynh_script_progression --message="Configuring NGINX web server..." --weight=1
 
-# Set permissions to app files
-chown -R root: $final_path
-# Files owned by user specific can just read
-find $final_path -type f | xargs chmod 644
-find $final_path -type d | xargs chmod 755
-
-# Except for data and tmp subdir, where the user must have write permissions
-mkdir -p $final_path/{data,tmp}
-chown -R $app:root $final_path/{data,tmp}
-chmod 700 $final_path/{data,tmp}
+# Create a dedicated NGINX config
+ynh_add_nginx_config
 
+#=================================================
+# GENERIC FINALIZATION
 #=================================================
 # SETUP SSOWAT
 #=================================================
-ynh_script_progression --message="Configuring permissions..."  --weight=1
+ynh_script_progression --message="Configuring permissions..." --weight=1
 
-# If app is public, add url to SSOWat conf as skipped_uris
-if [ $is_public -eq 1 ]; then
-    # unprotected_uris allows SSO credentials to be passed anyway.
-    ynh_permission_update --permission="main" --add="visitors"
+# Make app public if necessary
+if [ $is_public -eq 1 ]
+then
+	# Everyone can access the app.
+	# The "main" permission is automatically created before the install script.
+	ynh_permission_update --permission="main" --add="visitors"
 else
-    # If the app is private, viewing paste stays publicly accessible.
-    # Modify the domain to be used in a regex
-    domain_regex=$(echo "$domain" | sed 's@-@.@g')
-    ynh_app_setting_set --app=$app --key=protected_regex --value="$domain_regex$path_url/$"
-    ynh_app_setting_set --app=$app --key=unprotected_regex --value="$domain_regex$path_url/.*$"
+	# If the app is private, viewing paste stays publicly accessible.
+	# Modify the domain to be used in a regex
+	domain_regex=$(echo "$domain" | sed 's@-@.@g')
+	ynh_app_setting_set --app=$app --key=protected_regex --value="$domain_regex$path_url/$"
+	ynh_app_setting_set --app=$app --key=unprotected_regex --value="$domain_regex$path_url/.*$"
```

Comment:
For the love of god we should really try to get rid of those ... I don't know how exactly to fix those (probably playing with the `--url` or `--additional_url` of the `ynh_permission_*` helper) but the whole thing about `(un)protected_regex` is super legacy and that hack with `$domain_regex` is unreliable and creates bug. Another solution is to remove the ability to host the app as private (or "semi-private" in that case).

Sooner or later the package linter is going to report this code as an error é_è. Only 5 apps (privatebin, bozon, jirafeau, lstu and lutim) still use (un)protected_regexes

Reply:
protexted_regex removed